content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+```
+
+Add the function @getMessages@ below the @sendMyMessage@ function to update the newsfeed with the new updates:
+
+```[javascript]
+function getMessages() {
+    setState(prevState => {
+      return {
+        ...prevState,
+        msgs: [...prevState.msgs, ...prevState.newMsgs].sort(
+          (a, b) => b.date - a.date
+        ),
+        newMsgs: []
+      };
+    });
+    scrollRef.current.scrollTop = 0;
+}
+```
+Now create a new file @UICard.js@ and add the following code:
+
+```[javascript]
+import React from 'react'
+import { Card, Image, Segment } from 'semantic-ui-react'
+
+function formatAMPM(date) {
+  var hours = date.getHours();
+  var minutes = date.getMinutes();
+  var ampm = hours >= 12 ? 'PM' : 'AM';
+  hours %= 12;
+  hours = hours ? hours : 12;
+  minutes = minutes < 10 ? '0' + minutes : minutes;
+  var strTime = hours + ':' + minutes + ' ' + ampm;
+  return strTime;
+}
+
+const UICard = (props) => {
+  return (
+    <Segment>
+{props.events.map((event, index) => {
+  return (
+    <Card fluid key={index} index={index}>
+      <Card.Content>
+        <Image
+          floated='right'
+          size='mini'
+          src={event.image}
+        />
+        <Card.Header>{event.name}</Card.Header>
+        <Card.Meta>{formatAMPM(event.date)}</Card.Meta>
+        <Card.Description>
+          {event.summary}
+        </Card.Description>
+      </Card.Content>
+    </Card>
+     )
+    })}
+  </Segment>
+)
+  }
+
+export default UICard
+```
+
+So, the @formatAMPM@ function is used to display to date in readable format. Using the props the @UICard@ component, displays the newsfeed.
+
+Now, let’s update the @return@ of the @App@ component to use @UICard@ component to display the newsfeed with the following code:
+
+```[javascript]
+return (
+    <Container textAlign="left" className="App-header">
+      <Header as="h2">Ably React Tutorial - RealTime</Header>
+      <Form>
+        <TextArea
+          value={todoInput}
+          placeholder="Post an update"
+          onChange={e => setTodoInput(e.target.value)}
+        />
+        <div className="App-button">
+          <Button content="Send" primary onClick={sendMyMessage} />
+        </div>
+      </Form>
+
+      <Divider />
+      <div onClick={getMessages} className="App-update">
+        {state.newMsgs.length !== 0 && (
+          <Label>
+            {state.newMsgs.length.toString()} new update
+            {state.newMsgs.length === 1 ? "" : "s"} have arrived!
+          </Label>
+        )}
+      </div>
+      {state.msgs.length !== 0 && (
+        <div ref={scrollRef} className="App-cards">
+          <UICard events={state.msgs} />
+        </div>
+      )}
+    </Container>
+  );
+```  
+
+"See this step in Github":https://github.com/ably/tutorials
+
+h2. Step 5 - Getting online users using Ably Channel Occupancy API
+
+Add a new @metaChannel@ variable to get meta information. Add it below the @presenceChannel@ variable:
```

Comment:
The reader has no idea what a meta channel is. You'll need to add explanations throughout, please.